pkg/contextstore/metadata_test.go

```diff
@@ -0,0 +1,110 @@
+package contextstore
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+var testMetadata = ContextMetadata{
+	Metadata: map[string]interface{}{
+		"baz": "foo",
+	},
+	Endpoints: map[string]EndpointMetadata{
+		"test-ep": {
+			"foo": "bar",
+		},
+	},
+}
+
+func TestMetadataMarshalling(t *testing.T) {
+	var ctxMeta ContextMetadata
+	bytes, err := json.Marshal(&testMetadata)
+	assert.NilError(t, err)
+	err = json.Unmarshal(bytes, &ctxMeta)
+	assert.NilError(t, err)
+	assert.DeepEqual(t, ctxMeta, testMetadata)
+}
```

Comment:
Is this really just a `encoding/json` unit test, or is there something specific to `ContextMetadata` which you worry might not work?

Reply:
At the beginning it was just a personal check about how encoding/json deals with aliased types. Removing that !